src/libsyntax/parse/attr.rs

```diff
@@ -18,8 +18,21 @@ const DEFAULT_UNEXPECTED_INNER_ATTR_ERR_MSG: &str = "an inner attribute is not \
                                                      permitted in this context";
 
 impl<'a> Parser<'a> {
+    crate fn parse_arg_attributes(&mut self) -> PResult<'a, Vec<ast::Attribute>> {
+        self.do_parse_outer_attributes(|s, a| if !a.is_sugared_doc {
```

Comment:
Doc comments on fn parameters should also be feature gated, so their spans need to be collected, so this condition is incorrect.

Also, you can collect the spans from the `Vec` result of `parse_outer_attributes` instead of introducing the closure.

Reply:
If I do so, it will effectively kill the `documentation comments cannot be applied to method arguments` error message and instead will show `allow, cfg, cfg_attr, deny, forbid, and warn are the only allowed built-in ...`.

Reply:
Why?
It will add the "attributes on formal function parameters are unstable" message, yes (which you can specialize for doc comments), but AST validation will still say "documentation comments cannot ...".

Reply:
I am sorry, it was a misunderstanding. Nevertheless, I can't see the point of feature gating doc comments. Even for specialized messages like `doc comments on function parameters are unstable` doesn't make any sense because they aren't permitted in the first place

Reply:
Since they don't produce an error during parsing, they are permitted if they never reach AST validation, for example
```rust
#[cfg(FALSE)]
fn f(/** text */ x: u8) {}

macro_rules! blackhole { ($item:item) => () }

blackhole!(fn g(/** text */ x: u8) {});
```
(This can be added as a test, btw.)